source/connect/csot.txt

```diff
@@ -0,0 +1,234 @@
+~~~~~~~~~~~~~~~~~~~
+
+When you specify a ``timeout_ms`` option, the driver applies the timeout
+according to the same inheritance behaviors as the other {+driver-short+} options.
+The following table describes how the timeout value is inherited at each level:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Level
+     - Inheritance Description
+
+   * - Operation
+     - Takes the highest precedence and will override ``timeout_ms``
+       options set at any other level.
+
+   * - Transaction
+     - Takes precedence over ``timeout_ms`` set at the session,
+       collection, database, or client level.
+
+   * - Session
+     - Applies to all transactions and operations within
+       that session, unless the option is overridden by options set at those levels.
+
+   * - Database
+     - Applies to all sessions and operations within that
+       database, unless the option is overridden by options set at those levels.
+
+   * - Collection
+     - Applies to all sessions and operations on that
+       collection, unless the option is overridden by options set at those levels.
+
+   * - Client
+     - Applies to all databases, collections, sessions, transactions, and
+       operations within that client that do not otherwise specify
+       ``timeout_ms``.
+
+For more information on overrides and specific options, see the :ref:`Overrides
+<ruby-csot-overrides>` section.
+
+.. _ruby-csot-overrides:
+
+Overrides
+---------
+
+You can specify a ``timeout_ms`` option at the operation level to override the
+client-level configuration for a specific operation. This allows you to
+customize timeouts based on the needs of individual queries.
+
+The following example demonstrates how an operation-level ``timeout_ms``
+configuration can override a client-level ``timeout_ms`` configuration:
+
+.. literalinclude:: /includes/connect/csot.rb
+   :language: ruby
+   :start-after: start-csot-overrides
+   :end-before: end-csot-overrides
+
+Transactions
+~~~~~~~~~~~~
+
+When you create a new ``Mongo::Session`` instance to implement a transaction, use
+the ``default_timeout_ms`` option. You can set ``default_timeout_ms`` to specify the
+``timeout_ms`` values to use for:
+
+- `commit_transaction <{+api-root+}/Mongo/Session.html#commit_transaction-instance_method>`__
+- `abort_transaction <{+api-root+}/Mongo/Session.html#abort_transaction-instance_method>`__
+- `with_transaction <{+api-root+}/Mongo/Session.html#with_transaction-instance_method>`__
+- `end_session <{+api-root+}/Mongo/Session.html#end_session-instance_method>`__
+
+If you do not specify ``default_timeout_ms``, the driver uses the ``timeout_ms``
+value set on the parent ``Mongo::Client``.
+
+You cannot override the ``timeout_ms`` value of the ``Mongo::Client`` for a 
+call to ``start_session``. 
+
+You can only set a timeout value for the 
+`start_transaction <{+api-root+}/Mongo/Session.html#start_transaction-instance_method>`__
+method by using the ``timeout_ms`` option.
```

Comment:
For tech reviewer, can you confirm that this statement is true? Thank you!

Reply:
Are you asking to confirm that `default_timeout_ms` is not applied to `start_transaction`? If so, yes, that is true. Note that `start_transaction` doesn't actually perform any operations on the server, but only applies any options  given to it, to subsequent transaction-related commands.